doc/architecture/software-update.rst

```diff
@@ -17,53 +17,128 @@ Ostro OS based software can be deployed to a target device in two ways:
 - Software Update
 
   This is a component that Ostro OS borrows from Clear Linux\* OS
-  for Intel |reg| Architecture and consists of both a server and client
-  component.
-
+  and consists of both a server and client component.
+  While Clear Linux is focused on Intel |reg| Architecture, the
+  Software Update component is not tied to any specific platform
+  nor architecture, as long as the concept of rootfs/filesystem is
+  supported.
+
+
+Software Update
+===============
+The main concept behind the `Software Update tool`_ is that it doesn't
+rely on the target device to perform the installation of packages,
+handle file conflicts, dependency and versioning check.
+Instead, all of this happens a priori, on a build server.
+And it happens only once, instead of trying to repeat the same
+operations on each device.
+All the device needs to do is to lookup a reference file (manifest)
+for the list of files it must acquire, should they not be present locally.
+The tool focuses on high level functionality (bundles), instead of
+individual packages.
+In detail, the Software Update functionality is comprised of a pair of
+tools:
+
+ - Software Update Server
+
+   A back-end tool which generates the data stream for updates and bundles.
+
+ - Software Update Client
+
+   A tool that resides on the device and uses the data stream produced
+   by the Server.
+
+.. note::
+   The client needs an URL exposing the data stream.
+   This requires setting up a web server [ex: Apache, Nginx] pointing
+   to the "www" directory generated by the SW Update Server.
+   The SW Update server does *not* act as web server, it merely produces
+   data that the manager of the update stream will have to expose
+   through a separate tool.
 
 Software Update Server
 ======================
 
-The software update server runs for example,
-on a CI (Continuous Integration) server and not on the device itself.
+The SW Update Server is a tool used to produce the stream of updates
+that, later on, a device can use to upgrade itself.
+The SW Update Server runs on the back-end machine(s) producing the build
+artefacts.
+For example, in the case of a large build infrastructure, it can run on
+a CI (Continuous Integration) server.
 For each build, the server generates information specific to that build
 (in software update lingo, the delta from version 0) and related to
 previous builds (deltas from a customizable number of previous versions).
 
-The information produced by the server is then provided by an HTTP server
-and consumed by the software update client running on supported devices.
+The information produced by the server is then exposed as URL by a web
+server and used by the software update client running on supported devices.
+
+The Server secures the update information by generating hashes of each
+file (in a manifest) and then signing the file containing the references
+to the hashes (signature of the manifest).
```

Comment:
I would prefer to explain it later.
In this case I can replace HTTP with "web" and leave the detailed explanation to the point where the web server is actually needed.